Kernel/VM/RingBuffer.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2021, Sahan Fernando <sahan.h.fernando@gmail.com>.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/StringView.h>
+#include <Kernel/PhysicalAddress.h>
+
+namespace Kernel {
+
+class RingBuffer {
+public:
+    RingBuffer(const StringView region_name, size_t capacity);
```

Comment:
Let's take the `region_name` as a `String` so we can `move()` it along to `allocate_contiguous_kernel_region`.

Reply:
Fixed in the next revision.